client/src/app/components/Autocomplete.tsx

```diff
@@ -0,0 +1,288 @@
+    setInputValue("");
+    setMenuIsOpen(false);
+  };
+
+  /** add the current input value as a chip */
+  const handleEnter = () => {
+    if (inputValue.length) {
+      addChip(inputValue);
+    }
+  };
+
+  const handleTab = () => {
+    if (menuItems.length === 3) {
+      setInputValue(menuItems[2].props.children);
+    }
+    setMenuIsOpen(false);
+  };
+
+  /** close the menu when escape is hit */
+  const handleEscape = () => {
+    setMenuIsOpen(false);
+  };
+
+  /** allow the user to focus on the menu and navigate using the arrow keys */
+  const handleArrowKey = () => {
+    if (menuRef.current) {
+      const firstElement = menuRef.current.querySelector<HTMLButtonElement>(
+        "li > button:not(:disabled)"
+      );
+      firstElement && firstElement.focus();
+    }
+  };
+
+  /** reopen the menu if it's closed and any un-designated keys are hit */
+  const handleDefault = () => {
+    if (!menuIsOpen) {
+      setMenuIsOpen(true);
+    }
+  };
+
+  /** enable keyboard only usage while focused on the text input */
+  const handleTextInputKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case "Enter":
+        handleEnter();
+        break;
+      case "Escape":
+        handleEscape();
+        break;
+      case "Tab":
+        handleTab();
+        break;
+      case "ArrowUp":
+      case "ArrowDown":
+        handleArrowKey();
+        break;
+      default:
+        handleDefault();
+    }
+  };
+
+  /** apply focus to the text input */
+  const focusTextInput = (closeMenu = false) => {
+    searchInputRef.current?.querySelector("input")?.focus();
+    closeMenu && setMenuIsOpen(false);
+  };
+
+  /** add the text of the selected item as a new chip */
+  const onSelect = (event?: React.MouseEvent<Element, MouseEvent>) => {
+    if (!event) {
+      return;
+    }
+    const selectedText = (event.target as HTMLElement).innerText;
+    addChip(selectedText);
+    event.stopPropagation();
+    focusTextInput(true);
+  };
+
+  /** close the menu when a click occurs outside of the menu or text input group */
+  const handleClick = (event?: MouseEvent) => {
+    console.log("click", event);
+    if (!event) {
+      return;
+    }
+    if (searchInputRef.current?.contains(event.target as HTMLElement)) {
+      setMenuIsOpen(true);
+    } else if (
+      menuRef.current &&
+      !menuRef.current.contains(event.target as HTMLElement) &&
+      searchInputRef.current &&
+      !searchInputRef.current.contains(event.target as HTMLElement)
+    ) {
+      setMenuIsOpen(false);
+    }
+  };
+
+  /** enable keyboard only usage while focused on the menu */
+  const handleMenuKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case "Tab":
+      case "Escape":
+        event.preventDefault();
+        focusTextInput();
+        setMenuIsOpen(false);
+        break;
+      case "Enter":
+      case " ":
+        setTimeout(() => setMenuIsOpen(false), 0);
```

Comment:
to allow onSelect to perform as well. although this isn't needed anymore as i've changed to close on selection (because otherwise the user wouldn't even see the label being added).